File: go/217.go
/*
Use a map
*/
func containsDuplicate(nums []int) bool {
    set := make(map[int]int)
    for _, val := range nums {
        if _, ok := set[val]; ok {
            return true
        } else {
            set[val] = 1
        }
    }
    return false
}

File: solution/python/217.py
# Use dictionary to store the values. Or use set() alternatively.
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        numsDict = {}
        for i in nums:
            if i not in numsDict:
                numsDict[i] = 1
            else:
                return True
        return False